src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -98,17 +105,53 @@ def run_python_test(transitive_hydrated_target, pytest, python_setup):
   requirements_pex_response = yield Get(
     ExecuteProcessResult, ExecuteProcessRequest, requirements_pex_request)
 
-  # Gather sources.
+  source_roots = source_root_config.get_source_roots()
+
+  # Gather sources and adjust for the source root.
   # TODO: make TargetAdaptor return a 'sources' field with an empty snapshot instead of raising to
   # simplify the hasattr() checks here!
+  # TODO: find some way to restore the full source name for the stdout of the Pytest run, so that
+  # we can output the original path, rather than the path without the source root. It appears we
+  # do this in V1 via parsing the Junit XML file and adding the value back? See pytest_run.py's
+  # _get_target_from_test(). This is a low priority.
```

Comment:
```suggestion
  # _get_target_from_test().
```
I think adding comments about priority in the code are subject to bitrot pretty quickly, personally.

Reply:
Fair. Removed and moved most the content to https://github.com/pantsbuild/pants/issues/7714.